sunbeam-python/sunbeam/features/tls/common.py

```diff
@@ -139,15 +140,39 @@ def post_enable(
         self, deployment: Deployment, config: TlsFeatureConfig, show_hints: bool
     ) -> None:
         """Handler to perform tasks after the feature is enabled."""
-        jhelper = JujuHelper(deployment.juju_controller)
-        plan = [
+        if deployment.region_ctrl_juju_controller:
+            # This is a secondary region, Keystone is expected to run in the
+            # primary region.
+            jhelper = JujuHelper(deployment.region_ctrl_juju_controller)
+        else:
+            jhelper = JujuHelper(deployment.juju_controller)
+        this_region_name = read_config(deployment.get_client(), REGION_CONFIG_KEY)[
+            "region"
+        ]
+        plan: list[BaseStep] = [
             AddCACertsToKeystoneStep(
                 jhelper,
-                self.ca_cert_name,
+                self.ca_cert_name(this_region_name),
                 config.ca,  # type: ignore
                 config.ca_chain,  # type: ignore
             )
         ]
+        if deployment.region_ctrl_juju_controller and deployment.juju_controller:
+            # Some of the JujuHelper methods ignore the specified controller and
```

Comment:
I head the reasoning from this comment, but we can't resurface the region_ctrl complexity everywhere because of a lookup bug.

If we could reduce this overall complexity of switching around everywhere, fixing this bug should be priority.